server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/GetMessageListMethodTest.java

```diff
@@ -81,6 +82,63 @@ public void setup() throws Throwable {
     public void teardown() {
         jmapServer.stop();
     }
+
+    @Test
+    public void getMessagesShouldBehaveAsIfMailboxDidNotExistWhenUsedOnAnOtherUserMailbox() throws Exception {
+        // Attacker have an account on James JMAP
+        String attacker = "attacker@" + domain;
+        String password = "password";
+        jmapServer.serverProbe().addUser(attacker, password);
+        AccessToken attackerAccessToken =  JmapAuthentication.authenticateJamesUser(attacker, password);
+
+        // Put something in INBOX of the victim
+        jmapServer.serverProbe().createMailbox(MailboxConstants.USER_NAMESPACE, username, "INBOX");
+        jmapServer.serverProbe().appendMessage(username, new MailboxPath(MailboxConstants.USER_NAMESPACE, username, "INBOX"),
+            new ByteArrayInputStream("Subject: test\r\n\r\ntestmail".getBytes()), new Date(), false, new Flags());
+        await();
+
+        // Attacker knows victim INBOX id
+        String victimInboxId = getInboxId(accessToken);
+
+        // Empty messageIds is return result when mailbox do not exist
+        // See getMessageListShouldFilterMessagesWhenInMailboxesFilterDoesntMatches
+        given()
+            .accept(ContentType.JSON)
+            .contentType(ContentType.JSON)
+            .header("Authorization", attackerAccessToken.serialize())
+            .body("[[\"getMessageList\", {\"filter\":{\"inMailboxes\":[\"" + victimInboxId + "\"]}}, \"#0\"]]")
+        .when()
+            .post("/jmap")
+        .then()
+            .statusCode(200)
+            .body(NAME, equalTo("messageList"))
+            .body(ARGUMENTS + ".messageIds", empty());
```

Comment:
The axact same as if the mailbox was not found. See **getMessageListShouldFilterMessagesWhenInMailboxesFilterDoesntMatches** specific test....